lib/api/client-commands/registerBasicAuth.js

```diff
@@ -0,0 +1,49 @@
+const ClientCommand = require('./_base-command.js');
+const {Logger} = require('../../utils');
+
+/**
+ * Automate the input of basic auth credentials whenever they arise.
+ *
+ * @example
+ *  this.demoTest = function (browser) {
+ *    browser.registerBasicAuth(test, test);
+ *    browser.url('http://browserspy.dk/password-ok.php');
+ *  };
+ *
+ *
+ * @method registerBasicAuth
+ * @param {string} username
+ * @param {string} password
+ * @param {function} [callback] Optional callback function to be called when the command finishes.
```

Comment:
we also need `@syntax` and `@api` tags. See https://github.com/nightwatchjs/nightwatch/blob/main/lib/api/protocol/url.js